### PR TITLE
chore: Capture and process order errors properly in Sentry

### PR DESCRIPTION
**Pull request by iamareebjamal**
For #4174

### CODE REVIEW

File: app/sentry.ts [hunk 1]
```diff
@@ -24,3 +31,25 @@ Sentry.init({
     },
     ...config.sentry
 });
+
+Sentry.configureScope(function(scope) {
+    function addAdapterError(error: any, event: Sentry.Event) {
```
Comment (niranjan94) [inline app/sentry.ts]: Codacy Issue found: [Expected indentation of 2 spaces but found 4.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=5114150)

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 5
- Added 24
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=5114150&bid=16814325)

File: app/sentry.ts [hunk 2]
```diff
@@ -1,17 +1,24 @@
...
 
 Sentry.init({
-    integrations: [new Ember()],
+    integrations: [
+        new Ember(),
+        new Dedupe(),
+        new CaptureConsole({
+            levels: ['error']
+        })
+    ],
```
Comment (niranjan94) [inline app/sentry.ts]: Codacy Issue found: [Expected indentation of 2 spaces but found 4.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=5114150)

File: app/sentry.ts [hunk 3]
```diff
@@ -24,3 +31,25 @@ Sentry.init({
...
     ...config.sentry
 });
+
+Sentry.configureScope(function(scope) {
+    function addAdapterError(error: any, event: Sentry.Event) {
+        if (error.isAdapterError) {
+            event.extra = {
+                ...event.extra,
+                adapter_errors: error.errors,
+                adapter_errors_json: JSON.stringify(error.errors)
```
Comment (niranjan94) [inline app/sentry.ts]: Codacy Issue found: [Expected indentation of 8 spaces but found 16.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=5114150)

File: app/sentry.ts [hunk 4]
```diff
@@ -1,17 +1,24 @@
...
     beforeSend(event: Sentry.Event) {
         const exception = event.exception?.values?.[0];
         const errorValue = exception?.value;
-        if (errorValue?.includes("Ember Data Request") && 
-            errorValue?.includes("404")) {
-            // Ignore 404 errors from Ember Data because 
-            // I don't know how to turn them off
-            return null;
+        if (errorValue?.includes("Ember Data Request")) {
+            if (errorValue?.includes("404")) {
```
Comment (niranjan94) [inline app/sentry.ts]: Codacy Issue found: [Strings must use singlequote.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=5114150)

File: app/sentry.ts [hunk 5]
```diff
@@ -24,3 +31,25 @@ Sentry.init({
...
+            event.extra = {
+                ...event.extra,
+                adapter_errors: error.errors,
+                adapter_errors_json: JSON.stringify(error.errors)
+            }
+        }
+    }
+
+    scope.addEventProcessor(function(event: Sentry.Event, hints: Sentry.EventHint) {
+        addAdapterError(hints.originalException, event);
```
Comment (niranjan94) [inline app/sentry.ts]: Codacy Issue found: [Expected indentation of 4 spaces but found 8.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=5114150)

File: app/sentry.ts [hunk 6]
```diff
@@ -24,3 +31,25 @@ Sentry.init({
...
 });
+
+Sentry.configureScope(function(scope) {
+    function addAdapterError(error: any, event: Sentry.Event) {
+        if (error.isAdapterError) {
+            event.extra = {
+                ...event.extra,
+                adapter_errors: error.errors,
+                adapter_errors_json: JSON.stringify(error.errors)
+            }
```
Comment (niranjan94) [inline app/sentry.ts]: Codacy Issue found: [Missing semicolon.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=5114150)

File: app/sentry.ts [hunk 7]
```diff
@@ -1,17 +1,24 @@
...
+    ],
     beforeSend(event: Sentry.Event) {
         const exception = event.exception?.values?.[0];
         const errorValue = exception?.value;
-        if (errorValue?.includes("Ember Data Request") && 
-            errorValue?.includes("404")) {
-            // Ignore 404 errors from Ember Data because 
-            // I don't know how to turn them off
-            return null;
+        if (errorValue?.includes("Ember Data Request")) {
```
Comment (niranjan94) [inline app/sentry.ts]: Codacy Issue found: [Strings must use singlequote.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=5114150)

File: app/sentry.ts (hunk 3 above)
Comment (niranjan94) [inline app/sentry.ts]: Codacy Issue found: [Missing space after key 'adapter_errors_json'.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=5114150)

File: app/sentry.ts [hunk 8]
```diff
@@ -24,3 +31,25 @@ Sentry.init({
     },
     ...config.sentry
 });
+
+Sentry.configureScope(function(scope) {
+    function addAdapterError(error: any, event: Sentry.Event) {
+        if (error.isAdapterError) {
+            event.extra = {
+                ...event.extra,
+                adapter_errors: error.errors,
```
Comment (niranjan94) [inline app/sentry.ts]: Codacy Issue found: [Missing space after key 'adapter_errors'.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=5114150)

File: app/sentry.ts [hunk 9]
```diff
@@ -24,3 +31,25 @@ Sentry.init({
...
+            }
+        }
+    }
+
+    scope.addEventProcessor(function(event: Sentry.Event, hints: Sentry.EventHint) {
+        addAdapterError(hints.originalException, event);
+
+        const args: any[] = event.extra?.arguments || [];
+        for (const arg of args) {
+            addAdapterError(arg, event);
```
Comment (niranjan94) [inline app/sentry.ts]: Codacy Issue found: [Expected indentation of 6 spaces but found 12.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=5114150)

File: app/sentry.ts [hunk 10]
```diff
@@ -24,3 +31,25 @@ Sentry.init({
...
+                adapter_errors_json: JSON.stringify(error.errors)
+            }
+        }
+    }
+
+    scope.addEventProcessor(function(event: Sentry.Event, hints: Sentry.EventHint) {
+        addAdapterError(hints.originalException, event);
+
+        const args: any[] = event.extra?.arguments || [];
+        for (const arg of args) {
```
Comment (niranjan94) [inline app/sentry.ts]: Codacy Issue found: [Expected indentation of 4 spaces but found 8.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=5114150)

File: app/sentry.ts [hunk 11]
```diff
@@ -24,3 +31,25 @@ Sentry.init({
     },
     ...config.sentry
 });
+
+Sentry.configureScope(function(scope) {
+    function addAdapterError(error: any, event: Sentry.Event) {
+        if (error.isAdapterError) {
+            event.extra = {
```
Comment (niranjan94) [inline app/sentry.ts]: Codacy Issue found: [Expected indentation of 6 spaces but found 12.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=5114150)

File: app/sentry.ts (hunk 8 above)
Comment (niranjan94) [inline app/sentry.ts]: Codacy Issue found: [Expected indentation of 8 spaces but found 16.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=5114150)

File: app/sentry.ts [hunk 12]
```diff
@@ -1,17 +1,24 @@
 import * as Sentry from '@sentry/browser';
-import { Ember } from '@sentry/integrations';
+import { CaptureConsole, Dedupe, Ember } from '@sentry/integrations';
 import config from 'open-event-frontend/config/environment';
 
 Sentry.init({
-    integrations: [new Ember()],
+    integrations: [
+        new Ember(),
```
Comment (niranjan94) [inline app/sentry.ts]: Codacy Issue found: [Expected indentation of 4 spaces but found 8.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=5114150)

File: app/sentry.ts (hunk 7 above)
Comment (niranjan94) [inline app/sentry.ts]: Codacy Issue found: [Expected indentation of 4 spaces but found 8.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=5114150)

File: app/sentry.ts [hunk 13]
```diff
@@ -24,3 +31,25 @@ Sentry.init({
     },
     ...config.sentry
 });
+
+Sentry.configureScope(function(scope) {
+    function addAdapterError(error: any, event: Sentry.Event) {
+        if (error.isAdapterError) {
+            event.extra = {
+                ...event.extra,
```
Comment (niranjan94) [inline app/sentry.ts]: Codacy Issue found: [Expected indentation of 8 spaces but found 16.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=5114150)

File: app/sentry.ts [hunk 14]
```diff
@@ -1,17 +1,24 @@
...
-import { Ember } from '@sentry/integrations';
+import { CaptureConsole, Dedupe, Ember } from '@sentry/integrations';
 import config from 'open-event-frontend/config/environment';
 
 Sentry.init({
-    integrations: [new Ember()],
+    integrations: [
+        new Ember(),
+        new Dedupe(),
+        new CaptureConsole({
```
Comment (niranjan94) [inline app/sentry.ts]: Codacy Issue found: [Expected indentation of 4 spaces but found 8.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=5114150)

File: app/sentry.ts [hunk 15]
```diff
@@ -24,3 +31,25 @@ Sentry.init({
     },
     ...config.sentry
 });
+
+Sentry.configureScope(function(scope) {
+    function addAdapterError(error: any, event: Sentry.Event) {
+        if (error.isAdapterError) {
```
Comment (niranjan94) [inline app/sentry.ts]: Codacy Issue found: [Expected indentation of 4 spaces but found 8.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=5114150)

File: app/sentry.ts (hunk 4 above)
Comment (niranjan94) [inline app/sentry.ts]: Codacy Issue found: [Expected indentation of 6 spaces but found 12.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=5114150)

File: app/sentry.ts [hunk 16]
```diff
@@ -24,3 +31,25 @@ Sentry.init({
...
+
+    scope.addEventProcessor(function(event: Sentry.Event, hints: Sentry.EventHint) {
+        addAdapterError(hints.originalException, event);
+
+        const args: any[] = event.extra?.arguments || [];
+        for (const arg of args) {
+            addAdapterError(arg, event);
+        }
+      return event;
+    });
```
Comment (niranjan94) [inline app/sentry.ts]: Codacy Issue found: [Expected indentation of 2 spaces but found 4.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=5114150)

File: app/sentry.ts [hunk 17]
```diff
@@ -24,3 +31,25 @@ Sentry.init({
...
+        if (error.isAdapterError) {
+            event.extra = {
+                ...event.extra,
+                adapter_errors: error.errors,
+                adapter_errors_json: JSON.stringify(error.errors)
+            }
+        }
+    }
+
+    scope.addEventProcessor(function(event: Sentry.Event, hints: Sentry.EventHint) {
```
Comment (niranjan94) [inline app/sentry.ts]: Codacy Issue found: [Expected indentation of 2 spaces but found 4.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=5114150)

File: app/sentry.ts [hunk 18]
```diff
@@ -1,17 +1,24 @@
 import * as Sentry from '@sentry/browser';
-import { Ember } from '@sentry/integrations';
+import { CaptureConsole, Dedupe, Ember } from '@sentry/integrations';
 import config from 'open-event-frontend/config/environment';
 
 Sentry.init({
-    integrations: [new Ember()],
+    integrations: [
+        new Ember(),
+        new Dedupe(),
```
Comment (niranjan94) [inline app/sentry.ts]: Codacy Issue found: [Expected indentation of 4 spaces but found 8.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=5114150)

File: app/sentry.ts [hunk 19]
```diff
@@ -24,3 +31,25 @@ Sentry.init({
...
+                adapter_errors: error.errors,
+                adapter_errors_json: JSON.stringify(error.errors)
+            }
+        }
+    }
+
+    scope.addEventProcessor(function(event: Sentry.Event, hints: Sentry.EventHint) {
+        addAdapterError(hints.originalException, event);
+
+        const args: any[] = event.extra?.arguments || [];
```
Comment (niranjan94) [inline app/sentry.ts]: Codacy Issue found: [Expected indentation of 4 spaces but found 8.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=5114150)

File: app/sentry.ts [hunk 20]
```diff
@@ -1,17 +1,24 @@
...
 import config from 'open-event-frontend/config/environment';
 
 Sentry.init({
-    integrations: [new Ember()],
+    integrations: [
+        new Ember(),
+        new Dedupe(),
+        new CaptureConsole({
+            levels: ['error']
+        })
```
Comment (niranjan94) [inline app/sentry.ts]: Codacy Issue found: [Expected indentation of 4 spaces but found 8.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=5114150)

**Comment by iamareebjamal** [comment]
It is as if someone deliberately told devs to not use logging and error reporting in both frontend and server. Nothing gets logged, even on critical errors, and only unhandled errors are reported, because it is not possible to suppress them, or else I think they would be suppressed as well

**Pull request merged**